### PR TITLE
Parser

### PR DESCRIPTION
**Pull request by Daniyel03**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 CC=clang
-CFLAGS = -g -Wall -Wextra -Werror -I./includes -I./includes/42get_next_line-master
+CFLAGS = -g  -gdwarf-3 -O0 -Wall -Wextra -Werror -I./includes -I./includes/42get_next_line-master
```
Comment (Daniyel03) [inline Makefile:2]: will be removed :sunglasses:

File: libft/ft_strjoin.c [hunk 2]
```diff
@@ -12,16 +12,16 @@
...
-	result = (char *)malloc(size * sizeof(char));
-	if (!result)
-		return (result);
-	ft_strlcpy(result, s1, size);
-	ft_strlcat(result, s2, size);
-	return (result);
-}
+// 	size = ft_strlen(s1) + ft_strlen(s2) + 1;
+// 	result = (char *)malloc(size * sizeof(char));
+// 	if (!result)
```
Comment (Daniyel03) [inline libft/ft_strjoin.c:22]: using strjoin from gnl :face_with_open_eyes_and_hand_over_mouth:

File: parser copy.c [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+#include "../includes/cub3d.h"
+#include "../includes/parser.h"
+
+// void	free_parser(t_parser *parser)
+// {
+// 	if (parser->fd) || -1
+// 		close (parser->fd);
+// 	if (parser->parser->temp_fd)
+// 		close (parser->parser->temp_fd);
+// 	if (parser->last_graphic)
```
Comment (Daniyel03) [inline parser copy.c:22]: cleanup

File: src/parser.c [hunk 4]
```diff
@@ -6,146 +6,407 @@
...
+	parser->number = temp;
+	temp = NULL;
+}
+
+void	set_digits(t_parser *parser)
+{
+	parser->number = ft_strdup("0x");
+	parser->ptr = parser->numbers;
+	while (*(char **)parser->ptr)
+	{
```
Comment (Daniyel03) [inline src/parser.c:141]: how to iterate @Han2-Ro

Comment (Han2-Ro) [reply src/parser.c:141]: cool

**Pull request merged**